docs/v1.md

```diff
@@ -20,29 +21,35 @@
 
 ## Bindings
 - all() : `Binding[]`
+- extensions() : `Extensions`
+- filterBy(filter : `(binding: Binding) => boolean`) : `Binding[]`
 - get(id: `string`) : `Binding` | `undefined`
 - has(id: `string`) : `boolean`
 - isEmpty() : `boolean`
 
-## Channels
-- all() : `Channel[]`
-- filterBy(filter : `(channel: Channel) => boolean`) : `Channel[]`
-- filterBySend() : `Channel[]`
-- filterByReceive() : `Channel[]`
-- get(id: `string`) : `Channel` | `undefined`
-- has(id: `string`) : `boolean`
-- isEmpty() : `boolean`
-
 ## Channel
+- address() : `string`
 - bindings() : `Bindings`
 - description() : `string` | `undefined`
 - extensions() : `Extensions`
 - hasDescription() : `boolean`
+- id() : `string`
 - messages() : `Messages`
 - operations() : `Operations`
 - parameters() : `ChannelParameters`
 - servers() : `Servers`
 
+## Channels
+- all() : `Channel[]`
+- filterBy(filter : `(channel: Channel) => boolean`) : `Channel[]`
+- filterBySend() : `Channel[]`
+- filterByReceive() : `Channel[]`
+- filterByInUse() : `Channel[]`
+- filterByNotInUse() : `Channel[]`
```

Comment:
In v3, so far, a channel may be:

1. Defined but not used.
2. Defined and used by an operation.
3. Defined under components and not used.
4. Defined under components and used. (which is weird, I have to review it again)

The fact that something is under the `components` keyword, [makes it optional](https://github.com/asyncapi/spec/pull/803/files) by default.

What about we provide yet another method like `filterByOptional`? And its variations, `filterByOptionalInUse` and `filterByOptionalNotInUse` (or other names).

This would also remove the need of having a `components().channels()` function. And we can probably remove the need of all the components objects like this. What do you think?

Reply:
`InUse` and `NotInUse` determines which objects are actively (defined, referenced at least once outside of components, but still to components can belong), so the 3rd case will be `NotInUse`, and 1,2 and 4 to `InUse`.

Reply:
Yeah, I get it. I mean that to get those that are defined inside `components` without explicitly mentioning `components`, we could add the `filterByOptional` method. This will return those in the `components` section. Note how I'm referring to those by their meaning (optional) and not their location (components). This is not about `InUse` or `NotInUse`, it's about optional or not optional. An optional element (those inside components) may or may not be "in use".

Reply:
Ok, I also understand your pov, thanks! However next to the optional I think we should still have `inUse` and `notInUse` and by concatenating `filterByOptional` (or by `filterByInComponents` - maybe it's more verbose method)  we can execute this: `asyncapi.messages().filterByOptional().filterByInUse()` - you have all methods which are defined in components and are used in some places in channels/operations. WDYT?

Reply:
Alright, you just came to the same conclusion that I described above 😄 

> What about we provide yet another method like `filterByOptional`? And its variations, `filterByOptionalInUse` and `filterByOptionalNotInUse` (or other names).

Reply:
Sorry I forgot about that first comment, I only read the second one 😆 So I agree with you. @smoya @jonaslagoni wdyt?

Reply:
Sounds good 👍

Reply:
I'm not sure if I'm super biased by current Spec already... But trying to think on the UX i find that the "optional" concept sounds hard to understand at a first glance.
I think is because we mix what they (objects located under components) mean from the point of view of the spec (optional), but also from the point of view of the application (in use/not in use).

After giving another though, I think the word `component` is being misused here: It is not a place but a [model](https://github.com/asyncapi/parser-api/#models) (not added into `parser-api` README yet).
Having a proper description of what a component is, like in https://github.com/asyncapi/spec/pull/803 is more than enough to keep using `Components` as something an AsyncAPI document will keep having probably in mid/long term. Components, those "reusable objects for different aspects of the AsyncAPI specification" that are optional by default.

My suggestion is that we keep the `root.Components()` method and just add the `components.inUse` and `components.notInUse()` methods.

If I'm missing a point, have mercy on me as it's a bit late on my place 😆

Reply:
Yeah, you're right. It's gonna make it more complicated and there's a low risk the concept of "components" disappears or gets renamed.

Reply:
FYI my suggestion is on a PR here https://github.com/asyncapi/parser-js/pull/619